homeassistant/components/azuredns/__init__.py

```diff
@@ -0,0 +1,142 @@
+"""Support for Azure DNS services."""
+import logging
+from datetime import timedelta
+
+import voluptuous as vol
+
+import homeassistant.helpers.config_validation as cv
+from homeassistant.const import CONF_HOST, CONF_DOMAIN
+from homeassistant.helpers.event import async_track_time_interval
+from homeassistant.helpers.aiohttp_client import async_get_clientsession
+
+REQUIREMENTS = ['adal==1.2.1']
+
+_LOGGER = logging.getLogger(__name__)
+
+DOMAIN = 'azuredns'
+
+INTERVAL = timedelta(minutes=5)
+
+AUTHORITYHOSTURL = 'https://login.microsoftonline.com'
+RESOURCE = 'https://management.azure.com/'
+
+CONF_CLIENTID = 'clientid'
+CONF_CLIENTSECRET = 'clientsecret'
+CONF_TENANT = 'tenant'
+CONF_SUBSCRIPTIONID = 'subscriptionid'
+CONF_RESOURCEGROUPNAME = 'resourcegroupname'
+
+CONFIG_SCHEMA = vol.Schema({
+    DOMAIN: vol.Schema({
+        vol.Required(CONF_DOMAIN): cv.string,
+        vol.Optional(CONF_HOST, default='@'): cv.string,
+        vol.Required(CONF_TENANT): cv.string,
+        vol.Required(CONF_CLIENTID): cv.string,
+        vol.Required(CONF_CLIENTSECRET): cv.string,
+        vol.Required(CONF_SUBSCRIPTIONID): cv.string,
+        vol.Required(CONF_RESOURCEGROUPNAME): cv.string,
+    })
+}, extra=vol.ALLOW_EXTRA)
+
+
+async def async_setup(hass, config):
+    """Initialize the Azure DNS component."""
+    domain = config[DOMAIN][CONF_DOMAIN]
+    host = config[DOMAIN][CONF_HOST]
+    resource = (RESOURCE)
+    tenant = config[DOMAIN][CONF_TENANT]
+    clientid = config[DOMAIN][CONF_CLIENTID]
+    clientsecret = config[DOMAIN][CONF_CLIENTSECRET]
+    authority_url = (AUTHORITYHOSTURL + '/' + config[DOMAIN][CONF_TENANT])
+    apiurl = ('https://management.azure.com/subscriptions/'
```

Comment:
Can you change that to `api_url`?

Reply:
Yes, this was changed with commit 7ce487787e67e4a8848772841ef0be26e5d54beb.